src/Hedgehog/Extras/Test/MonadAssertion.hs

```diff
@@ -30,3 +40,37 @@ instance MonadAssertion m => MonadAssertion (IO.ResourceT m) where
   catchAssertion r h = IO.ResourceT $ \i -> IO.unResourceT r i `catchAssertion` \e -> IO.unResourceT (h e) i
 
 deriving instance Monad m => MonadAssertion (H.PropertyT m)
+
+-- | Attempt to run a function that may assert, returning either a failure or the result of the assertion.
+tryAssertion :: ()
+  => MonadAssertion m
+  => m a
+  -> m (Either H.Failure a)
+tryAssertion m =
+  catchAssertion (Right <$> m) (pure . Left)
+
+-- | Run the given action and succeed if the action fails, but fail if it succeeds.
+assertFailure :: ()
```

Comment:
That replaces `expectFailure`, no? If yes, we should remove one. The similar case for `expectFailureWith` - we should put them together and use the same naming convention

Reply:
Maybe we should mark as deprecated some of them?